project1/client/client_tasks/download_task.h

```diff
@@ -0,0 +1,34 @@
+#ifndef PROJECT1_DOWNLOAD_TASK_H
+#define PROJECT1_DOWNLOAD_TASK_H
+
+#include "../../thread_pool/task.h"
+#include "../../wire_protocol/wire_protocol.h"
+#include "../client_util.h"
+
+namespace client_tasks {
+class DownloadTask : public thread_pool::Task {
+ public:
+  DownloadTask(const std::string& filename, std::vector<uint8_t> &incoming_buf,
```

Comment:
(minor) Generally good to avoid non-const references as parameters, because it's not obvious to the callee that the parameters can be modified. If these parameters cannot be const, pass them by pointer instead.

Reply:
Admittedly, it was initially const so I could get the code to compile and function like I wanted.